mewarpx/mewarpx/utils_store/init_restart_util.py

```diff
@@ -61,15 +61,16 @@ def get_sorted_checkpoints(checkpoint_directory, checkpoint_prefix):
 
 def clean_old_checkpoints(checkpoint_directory="diags",
                           checkpoint_prefix=DEFAULT_CHECKPOINT_NAME,
-                          num_to_keep=1):
+                          num_to_keep=2):
```

Comment:
I think we can still keep this as 1 since the old checkpoint is only deleted once the new one is completely written anyway. The new approach doesn't specifically need 2 checkpoints, it just needs at least 1 valid one.
```suggestion
                          num_to_keep=1):
```

Reply:
Yeah, but this falls into my concern about it being cautious/fail-safe rather than fail-bad: If we only keep 1 checkpoint, and then say (to use the same example) someone accidentally moves the fluxdata.ckpt file, that 1 checkpoint will be deleted on restart because it's assumed corrupt, and then you're done for - you've lost the whole run progress no matter what. By keeping 2, it makes the whole system robust against any of those weird circumstances where we don't want to auto-delete a checkpoint.

Reply:
Can you give me time to look over this tomorrow afternoon before merging it? I can probably identify a lot of other returnable edge cases, and figure out whether they're covered

Reply:
I just looked at where this is getting called, in `CheckPointDiagnostic`. That keeps its own member `num_to_keep` which it passes to this. It defaults to 1. I think that's a serious problem, because the 1 that it will keep will be the one that it just wrote. Since that write may not be immediately durable, it's definitely worth keeping its predecessor (i.e. at least 2). Given that, let's fix that, and also leave the default argument here at 2.